cypress/utils/utils.ts

```diff
@@ -1452,6 +1452,17 @@ export function selectWithinModal(selector: string): void {
     clickWithin(modal, selector);
 }
 
+/**
+ * Executes a function inside a specified HTML element
+ * @param selector parent element
+ * @param functionToExec function to execute
+ */
+export function callWithin(selector: string, functionToExec: () => void): void {
+    cy.get(selector)
+        .eq(0)
+        .within(() => functionToExec());
+}
+
```

Comment:
Let's modify this function so that it accepts index as a parameter.
export function callWithin(selector: string, functionToExec: () => void, index = 0): void {
    cy.get(selector)
           .eq(index)
          .within(() => functionToExec());
}